near-sdk/src/environment/hash.rs

```diff
@@ -36,9 +32,14 @@ impl CryptoHasher for Sha256 {
     fn hash(ingest: &[u8]) -> Self::Digest {
         unsafe { sys::sha256(ingest.len() as _, ingest.as_ptr() as _, ATOMIC_OP_REGISTER) };
 
-        let mut hash = [MaybeUninit::uninit(); 32];
-        read_register_fixed(ATOMIC_OP_REGISTER, &mut hash);
-        unsafe { std::mem::transmute(hash) }
+        let mut hash = [MaybeUninit::<u8>::uninit(); 32];
+        //* SAFETY: sha256 syscall will always generate 32 bytes inside of the atomic op register
```

Comment:
:+1: for adding a `SAFETY` section. Not sure I've seen `//*` sytnax before, usually it's done without it:

https://github.com/matklad/rust/blob/6063405c1b502f1c72262278e3b710854c10be25/library/alloc/src/alloc.rs#L206-L211

no strong opinion here.

Reply:
<img width="665" alt="Screen Shot 2021-11-17 at 11 30 03 AM" src="https://user-images.githubusercontent.com/24993711/142192639-5cf408d2-429c-4384-aa95-597ee9e01b0d.png">

My preference comes from seeing it in some codebases and it stands out more than regular docs, especially if you have an extension that changes the colour of different doc types, as shown in the screenshot above. I've used this style across other examples in the SDK and am happy to change, but everything should probably be consistent.